### PR TITLE
GCC machine descriptions

### PR DESCRIPTION
**Pull request by larsbrinkhoff**
The `.md` file extension is used not only for markdown files, but also GCC machine descriptions.  There are about [111000](https://github.com/search?q=extension%3Amd+insn) of those, so this language should qualify for addition.

The syntax is a Lisp-like variation of RTL, Register Transfer Language.

My intention is to open this issue for discussion, and later turn it into a PR if there are no objections.

### CODE REVIEW

**Comment by larsbrinkhoff** [comment]
@arfon @pchaigno Please review.

The Bayesian classifier doesn't do well with a set of 1000 `.md` samples.  I added a heuristic: if a line begins with `;;` or `(define_`, it's almost certainly a GCC machine description.  Almost anything else is a Markdown file.

File: lib/linguist/languages.yml [hunk 1]
```diff
@@ -1182,6 +1182,15 @@ GAS:
   tm_scope: source.assembly
   ace_mode: assembly_x86
 
+GCC machine description:
```
Comment (pchaigno) [inline lib/linguist/languages.yml]: Should we title case the name? (We can't change it after; so whatever we do, we better be sure about it :stuck_out_tongue: )

**Comment by pchaigno** [comment]
> The Bayesian classifier doesn't do well with a set of 1000 `.md` samples.

Did you get a chance to test the classifier + heuristic on those samples?

File: lib/linguist/languages.yml (hunk 1 above)
Comment (Alhadis) [inline lib/linguist/languages.yml]: Please do, holy shit. :|

**Comment by larsbrinkhoff** [comment]
> Did you get a chance to test the classifier + heuristic on those samples?

Of course.  There were only Markdown files in that set, so I tuned the heuristic until there were no false positives for GCC *M*achine *D*escriptions.

File: lib/linguist/languages.yml (hunk 1 above)
Comment (larsbrinkhoff) [inline lib/linguist/languages.yml]: Ok, history rewritten.

Comment (Alhadis) [inline lib/linguist/languages.yml]: Close call. Every OCD coder like myself would've gone nuts over the discrepancy.

Comment (larsbrinkhoff) [inline lib/linguist/languages.yml]: Read these and weep: [IRC log](https://github.com/github/linguist/blob/5066f66dcd7be1f8d68cecd1f226e2b4c517963b/lib/linguist/languages.yml#L1612), [OpenRC runscript](https://github.com/github/linguist/blob/5066f66dcd7be1f8d68cecd1f226e2b4c517963b/lib/linguist/languages.yml#L2618), [Python traceback](https://github.com/github/linguist/blob/5066f66dcd7be1f8d68cecd1f226e2b4c517963b/lib/linguist/languages.yml#L3031), [Raw token data](https://github.com/github/linguist/blob/5066f66dcd7be1f8d68cecd1f226e2b4c517963b/lib/linguist/languages.yml#L3146).

**Comment by pchaigno** [comment]
Ooh, 100% then, nice! :tada:

**Comment by larsbrinkhoff** [comment]
Well, some unusual Markdown files fall through the heuristic and lands in the Bayesian classifier, which doesn't always get it right.

**Comment by larsbrinkhoff** [comment]
Ok, here are the results from a set ot 999 `.md` files.  (Don't know where number 1000 went.)

| Language | Detected | Actual |
| --- | --- | --- | --- |
| Markdown | 993 | 996 |
| GCC Machine Description | 6 | 0 |
| Unknown| 0 | [3](https://github.com/lisp/de.setf.asdf.x/tree/master/website/source/resources) |

**Comment by arfon** [comment]
@larsbrinkhoff - is this good to :ship:?

**Comment by Alhadis** [comment]
ship-it

**Comment by larsbrinkhoff** [comment]
@arfon Yes.

**Pull request merged**